### PR TITLE
single-home the vanes

### PR DESCRIPTION
**Pull request by joemfb**
The Costello to #915's Abbott. This removes all the multi-homing indirection in the state, state-machines, and moves of all the vanes. The closes thing to an exception is `%gall`'s `%deal` task, which takes a pair of ships, but uses them alternatively for bidirectional communication.

I've tested networking, syncing/merging, and cross-ship app pokes in a number of scenarios, but there's obviously still risk involved in making this many changes across this many places. Drive-by reviews of the area one knows best would be greatly appreciated.

### CODE REVIEW

**Comment by Fang-** [comment]
Quality commit style! I drove by jael and parts of clay real quick, it all seems sane. I don't expect much weirdness here anyway, because of how the multi-homing code in vanes was used in practice.

So good to finally have this cleaned up, this used to be so ugly.

**Comment by joemfb** [comment]
Much obliged, @Fang-!

**Comment by cgyarvin** [comment]
Amazing.

File: sys/vane/clay.hoon [hunk 1]
```diff
@@ -429,47 +428,44 @@
...
 ++  de                                                  ::  per desk
-  |=  {now/@da hen/duct raft}
-  |=  {{our/@p her/@p} syd/desk}
-  =*  ruf  +>+<+>
-  =+  ^-  {hun/(unit duct) rede}
-      =+  rom=(~(get by fat.ruf) her)
-      ?~  rom
+  |=  [our=ship now=@da hen=duct raft]
+  |=  [her=ship syd=desk]
+  ::  XX ruf=raft crashes in the compiler
```
[the remark this answers is not in the document]
Comment (joemfb) [reply sys/vane/clay.hoon]: Good eye! I meant to leave a comment here about this specifically. Adding a structure to a `=*` is a new feature, and I wonder if it's interacting poorly with advanced/unusual lark wing there. I tried to switch this to put the faces in the sample and `=,` the faces, but there were compiler errors with that too.

File: sys/vane/gall.hoon [hunk 2]
```diff
@@ -1392,8 +1390,7 @@
   ^+  [*(list move) ..^$]
   ~|  [%gall-take tea]
   ?>  ?=({@ ?($sys $use) *} tea) 
-  =+  our=(need (slaw %p i.tea))
```
[the remark this answers is not in the document]
Comment (joemfb) [reply sys/vane/gall.hoon]: Yeah, I believe it. I removed some parsing of wires (here and in %clay specifically), but I didn't change the serialization or shape of the wires. I was worried about too many moving parts in this PR. But maybe it'd be better to get all the changes in at once and then test thoroughly on a testnet? I'm feeling conflicted now ...

**Comment by cgyarvin** [comment]
No excessive moving parts!  If you can split it, split it...

**Comment by eglaysher** [comment]
Per Curtis, minimize moving parts.

File: sys/vane/ford.hoon [hunk 3]
```diff
@@ -6109,59 +6100,39 @@
...
       %build
     ::  perform the build indicated by :task
     ::
-    ::    First, we find or create the :ship-state for :our.task,
-    ::    modifying :state-by-ship as necessary. Then we dispatch to the |ev
-    ::    by constructing :event-args and using them to create :start-build,
-    ::    which performs the build. The result of :start-build is a pair of
-    ::    :moves and a mutant :ship-state. We update our :state-by-ship map
-    ::    with the new :ship-state and produce it along with :moves.
+    ::    First, we dispatch to the |ev by constructing :event-args and
```
Comment (belisarius222) [inline sys/vane/ford.hoon]: `|ev` -> `|per-event core`

Comment (belisarius222) [reply sys/vane/ford.hoon]: (the comment was stale when you got there, so feel free to ignore)

File: tests/sys/vane/ford.hoon [hunk 4]
```diff
@@ -788,14 +788,14 @@
...
       ::
       ^=  call-args
-        :*  duct=~[/one]  type=~  %build  ~nul  live=%.n
+        :*  duct=~[/one]  type=~  %build  live=%.n
             [%scry %c ren=%x rail=[[~nul %desk] /bar/foo]]
         ==
       ::
       ^=  moves
         :~  :*  duct=~[/one]  %pass
                 wire=/~nul/scry-request/cx/~nul/desk/~1234.5.6/foo/bar
```
Comment (belisarius222) [inline tests/sys/vane/ford.hoon]: At some point we should pull `our` out of the wires that Ford uses, too. I can do it in a later PR if you'd like.

**Comment by joemfb** [comment]
Thanks all, I'll open an issue to track the removal of identity from these wires.

**Pull request merged**